modules/change_password.php

```diff
@@ -0,0 +1,86 @@
+<?php
+$userId = isset($_SESSION['login_user_id']) ? $_SESSION['login_user_id'] : null;
+if (!$userId) {
+    header('Location:index.php');
+    exit();
+}
+
+$isSuccess = isset($_GET['success']) ? $_GET['success']: null;
+
+if (!empty($_POST)) {
+    $current = md5($_POST['current']);
+    $update = md5($_POST['update']);
+    $confirm = md5($_POST['confirming']);
+
+    $search_current_password = "SELECT password FROM users WHERE id = $userId LIMIT 1";
+    $result = $mysql->query($search_current_password);
+    $user = $result->fetch_array() ?? false;
+
+    $update_password = "UPDATE users SET password = '$update' WHERE id = $userId";
+    $errors = [];
+    
+    if ($update === $confirm) {
+        if ($current === $user['password']) {
+            try{       
+                    $result2 = $mysql->query($update_password) ?? false;
+                    header('Location: index.php?m=change_password&success=true');
+                    exit();        
+            } catch (Exception $e) {
+                array_push($errors, $e->getMessage());
+            }  
+        } else {
+            header('Location: index.php?m=change_password&error=type1');
+            exit();
+        }  
+    } else {
+        header('Location: index.php?m=change_password&error=type2');
+        exit();
+    }
+}
```

Comment:
You modification is correct, but it is complex, need to redirect and nested if-else. Try to read Hint for homework No05 and fix it :)